d2/src/main/java/com/linkedin/d2/xds/XdsToD2PropertiesAdaptor.java

```diff
@@ -221,11 +221,19 @@ public void onChanged(XdsClient.NodeUpdate update)
           }
           catch (PropertySerializationException e)
           {
+            _serviceEventBus.publishInitialize(serviceName, null);
+            // still notify event bus to avoid timeout in case some subscribers are waiting for the data
             LOG.error("Failed to parse D2 service properties from xDS update. Service name: " + serviceName, e);
           }
         }
       }
 
+      @Override
+      public void onDelete(String name)
+      {
+        _serviceEventBus.publishRemove(name);
+      }
```

Comment:
See above, this should be:
```suggestion
      @Override
      public void onDelete()
      {
        _serviceEventBus.publishRemove(serviceName);
      }
```